shell/packages/sandstorm-db/migrations.js

```diff
@@ -191,6 +160,79 @@ function verifyAllPgpSignatures() {
   });
 }
 
+function splitUserIdsIntoAccountIdsAndIdentityIds() {
+  var Crypto = Npm.require("crypto");
+  Meteor.users.find().forEach(function (user) {
+    var identity = {};
+    var serviceUserId;
+    if ("devName" in user) {
+      identity.service = "dev";
+      serviceUserId = user.devName;
+    } else if ("expires" in user) {
+      identity.service = "demo";
+      serviceUserId = user._id;
+    } else if (user.services && "google" in user.services) {
+      identity.service = "google";
+      if (user.services.google.email && user.services.google.verified_email) {
+        identity.verifiedEmail = user.services.google.email;
+      }
+      serviceUserId = user.services.google.id;
+    } else if (user.services && "github" in user.services) {
+      identity.service = "github";
+      identity.unverifiedEmail = user.services.github.email;
+      serviceUserId = user.services.github.id;
+    } else if (user.services && "emailToken" in user.services) {
+      identity.service = "emailToken";
+      identity.verifiedEmail = user.services.emailToken.email;
+      serviceUserId = user.services.emailToken.email;
+    }
+
+    identity.id = Crypto.createHash("sha256")
+        .update(identity.service + ":" + serviceUserId).digest("hex");
+
+    if (user.profile) {
+      if (user.profile.name) {
+        identity.name = user.profile.name;
+      }
+      if (user.profile.handle) {
+        identity.handle = user.profile.handle;
+      }
+      if (user.profile.picture) {
+        identity.picture = user.profile.picture;
+      }
+      if (user.profile.pronoun) {
+        identity.pronoun = user.profile.pronoun;
+      }
+      if (user.profile.email) {
+        identity.unverifiedEmail = user.profile.email;
+      }
+    }
+    identity.main = true;
+
+    Meteor.users.update(user._id, {$set: {identities: [identity]}});
+
+    Grains.update({userId: user._id}, {$set: {identityId: identity.id}}, {multi: true});
+    Sessions.update({userId: user._id}, {$set: {identityId: identity.id}}, {multi: true});
+    ApiTokens.update({userId: user._id},
+                     {$set: {identityId: identity.id}},
+                     {multi: true});
+    ApiTokens.update({"owner.user.userId": user._id},
+                     {$set: {"owner.user.identityId": identity.id}},
+                     {multi: true});
+    ApiTokens.update({"owner.grain.introducerUser": user._id},
+                     {$set: {"owner.grain.introducerIdentity": identity.id}},
+                     {multi: true});
+    ApiTokens.update({"requirements.permissionsHeld.userId": user._id},
+                     {$set: {"requirements.$.permissionsHeld.identityId": identity.id}},
```

Comment:
Oh shoot, this doesn't actually do what I thought it did. The "$" only gets the first matched element. See http://docs.mongodb.org/manual/reference/operator/update/positional/#up._S_

Note that I use the "$" operator elsewhere. I'm not sure that this would be the right place to comment about what it means.